CenterForOpenScience/osf.io
# [OSF-8569] Mod Meta Cleanup

## Purpose

This PR clean up some legacy code as a result of this [PR](https://github.com/CenterForOpenScience/osf.io/pull/7530).

## Changes

1. Remove view method for Dataverse widget, moved to dataverse/utils.py.
2. Remove view method for forward (redirection) widget, moved to forward/utils.py
3. Remove view method for wiki widget, moved to wiki/utils.py
3. Remove api/v1 routes for wiki, zotero and mendeley widgets.
4. Change `nodes.py` to reuse the `widget()` method for citation providers.

## QA Notes

Since this PR is a clean up, the view templates for the widgets are not modified but only how/where we pass data to those widgets. A simple pass over Dataverse, wiki, forward(redirection), zotero and mendeley to make sure they function should be sufficient.

## Side effects

None.

## Ticket

https://openscience.atlassian.net/browse/OSF-8569
